reactComponents/src/searchbar/redux/tokenSearchSlice.ts

```diff
@@ -182,8 +199,9 @@ export const tokenSearchSlice = createSlice({
       if (!action.payload.checked) {
         action.payload.networks.forEach((network) => {
           if (network.id === action.payload.networkName) {
-            network.exchanges.forEach((exhange) => {
-              if (state.exchangeMap[exhange]) state.exchangeMap[exhange] = false;
+            network.exchanges.forEach((exchange) => {
+              const isDuplicated = isDuplicatedExchange(exchange, action.payload.networks, state.networkMap);
```

Comment:
to make it more efficient you could call the `isDuplicatedExchange` function only when the current exchange is selected. So it would be something like:
```
if (state.exchangeMap[exchange]) {
  state.exchangeMap[exchange] = isDuplicatedExchange(exchange, action.payload.networks, state.networkMap);
}
```

Reply:
@rochaeb I don't think your suggestion is not correct. 
In this case when user selected `Ethereum` and `Polygon` networks and selected `Sushiswap` in exchanges. 
But when user wants to unselect `Sushiswap`, he/she should double click that exchange.

So `isDuplicatedExchange` needs to call in `setNetworkMap` action